packages/scheduler/src/forks/Scheduler.js

```diff
@@ -85,34 +85,12 @@ var isHostCallbackScheduled = false;
 var isHostTimeoutScheduled = false;
 
 // Capture local references to native APIs, in case a polyfill overrides them.
-const setTimeout = window.setTimeout;
-const clearTimeout = window.clearTimeout;
-const setImmediate = window.setImmediate; // IE and Node.js + jsdom
-
-if (typeof console !== 'undefined') {
-  // TODO: Scheduler no longer requires these methods to be polyfilled. But
-  // maybe we want to continue warning if they don't exist, to preserve the
-  // option to rely on it in the future?
-  const requestAnimationFrame = window.requestAnimationFrame;
-  const cancelAnimationFrame = window.cancelAnimationFrame;
-
-  if (typeof requestAnimationFrame !== 'function') {
-    // Using console['error'] to evade Babel and ESLint
-    console['error'](
-      "This browser doesn't support requestAnimationFrame. " +
-        'Make sure that you load a ' +
-        'polyfill in older browsers. https://reactjs.org/link/react-polyfills',
-    );
-  }
-  if (typeof cancelAnimationFrame !== 'function') {
-    // Using console['error'] to evade Babel and ESLint
-    console['error'](
-      "This browser doesn't support cancelAnimationFrame. " +
-        'Make sure that you load a ' +
-        'polyfill in older browsers. https://reactjs.org/link/react-polyfills',
-    );
-  }
-}
+const localSetTimeout =
+  typeof window !== 'undefined' ? window.setTimeout : setTimeout;
+const localClearTimeout =
+  typeof window !== 'undefined' ? window.clearTimeout : clearTimeout;
+const localSetImmediate =
+  typeof window !== 'undefined' ? window.setImmediate : setImmediate; // IE and Node.js + jsdom
```

Comment:
This will insta-crash non-browser environments that don't have `setImmediate` global.

I suggest this pattern:

```
let localSetTimeout = typeof setTimeout === 'function' ? setTimeout : undefined;
let localClearTimeout = typeof clearTimeout === 'function' ? clearTimeout : undefined;
let localSetImmediate = typeof setImmediate === 'function' ? setImmediate : undefined;
```

Reply:
I'd recommend not hoisting these up like this and instead move them into where they're used so that the variable always has a consistent type. No need for the "undefined" intermediate step.

Reply:
setTimeout and clearTimeout are not only used at the top level, so how would you handle that without adding a runtime check cost?